Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -78,20 +78,31 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
 
         let kSpacingBetweenItems: CGFloat = 20
 
-        let headerView = UIStackView()
-        self.headerView = headerView
+        self.headerView = UIView()
+        headerView.layoutMargins = UIEdgeInsets(top: 2, left: 8, bottom: 4, right: 8)
 
         super.init(transitionStyle: .scroll,
                    navigationOrientation: .horizontal,
                    options: [UIPageViewControllerOptionInterPageSpacingKey: kSpacingBetweenItems])
 
-        // needed for proper layout on iOS9/10
-        headerView.translatesAutoresizingMaskIntoConstraints = false
+        let headerStackView = UIStackView()
+        headerView.addSubview(headerStackView)
 
-        headerView.axis = .vertical
-        headerView.alignment = .center
-        headerView.addArrangedSubview(headerNameLabel)
-        headerView.addArrangedSubview(headerDateLabel)
+        headerStackView.axis = .vertical
+        headerStackView.alignment = .center
+        headerStackView.spacing = 0
+        headerStackView.distribution = .fillProportionally
+        headerStackView.addArrangedSubview(headerNameLabel)
+        headerStackView.addArrangedSubview(headerDateLabel)
+
+        headerStackView.autoPinEdge(toSuperviewMargin: .top, relation: .greaterThanOrEqual)
+        headerStackView.autoPinEdge(toSuperviewMargin: .right, relation: .greaterThanOrEqual)
+        headerStackView.autoPinEdge(toSuperviewMargin: .bottom, relation: .greaterThanOrEqual)
+        headerStackView.autoPinEdge(toSuperviewMargin: .left, relation: .greaterThanOrEqual)
+        headerStackView.setContentHuggingHigh()
+        NSLayoutConstraint.autoSetPriority(UILayoutPriorityDefaultLow) {
```

Comment:
Does this have to be low priority?

Reply:
It just has to be lower priority than the constraints to the superview margin - so anything less than `UILayoutPriorityRequired` would do.

Reply:
Oh wait, you're right, it doesn't need to be low priority.